testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.testkit
+
+import java.nio.file.Files
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.TimeUnit
+
+import akka.persistence.cassandra.testkit.CassandraLauncher
+import com.lightbend.lagom.scaladsl.persistence.cassandra.testkit.TestUtil
+import com.lightbend.lagom.scaladsl.server.{ LagomApplication, LagomApplicationContext, RequiresLagomServicePort }
+import org.apache.cassandra.io.util.FileUtils
+import org.slf4j.LoggerFactory
+import play.api.ApplicationLoader.Context
+import play.api.{ Configuration, Environment, Play }
+import play.core.DefaultWebCommands
+import play.core.server.{ Server, ServerConfig, ServerProvider }
+
+import scala.concurrent.Future
+import scala.util.Try
+import scala.util.control.NonFatal
+
+/**
+ * Support for writing functional tests for one service. The service is running in a server and in the test you can
+ * interact with it using its service client, i.e. calls to the service API.
+ *
+ * The server is ran standalone without persistence, pubsub or cluster features enabled. Cassandra is also disabled by
+ * default. If your service require either of these features you can enable them in the `Setup`.
+ *
+ * There are two different styles that can be used. It is most convenient to use [[ServiceTest.withServer()]], since it
+ * automatically starts and stops the server before and after the given block. When your test have several test
+ * methods, and especially when using persistence, it is faster to only [[ServiceTest.startServer()]] the server once in a
+ * before all tests hook, and then stop it in an after all test hook.
+ */
+object ServiceTest {
+
+  sealed trait Setup {
```

Comment:
Is there really any benefit to using a paired trait with an implementation class instead of just a case class by itself? It's just data and there's no way to create another implementation of a sealed trait right?